pkg/kubelet/apis/config/types.go

```diff
@@ -656,3 +661,16 @@ type MemorySwapConfiguration struct {
 	// +optional
 	SwapBehavior string
 }
+
+// RateLimitConfiguration is used for setting the rate limits for the kubelet APIs exposed
+// locally on the node as gRPC over UNIX-domain sockets, like the podresources API
+type RateLimitConfiguration struct {
+	// MaxFrequency  defines the maximum allowed frequency of the calls represented
+	// as number of calls per second, cumulating all the calls.
+	// Omitting the value or setting to zero means no limit.
+	MaxFrequency float64
```

Comment:
The kubelet serves lots of endpoints, many of which incur cost and make remote connections (pod exec/attach/logs, portforward, etc). Is there a reason we're targeting rate limits at a single one of those endpoints? Would be good to get an ack from kubelet leads that this config is coherent with the overall direction of kubelet serving.

Reply:
we discussed this item (and https://github.com/kubernetes/kubernetes/pull/115852#discussion_r1126606012 and https://github.com/kubernetes/kubernetes/pull/115852#discussion_r1126605828) in sig-node meeting on March 7 and we don't have a conclusion just yet. We agreed to continue with the discussion here and/or on sig-node channel (cc @SergeyKanzhelev @dchen1107 )